cli/src/config/aliases.toml

```diff
@@ -0,0 +1,13 @@
+[aliases]
+# Placeholder: added by user
+
+[revset-aliases]
+'trunk()' = '''
+latest(
+  remote_branches(exact:"main", exact:"origin") | 
+  remote_branches(exact:"master", exact:"origin") | 
+  remote_branches(exact:"root", exact:"origin") |
```

Comment:
"root" is not a branch name, it's a special symbol (and might become a function instead as we talked about somewhere). So use just `root` instead of the `remote_branches()` expression

Reply:
I was intending for a branch named root here instead of the root symbol. Would it make sense for this to default to root? For example, if we did `jj rebase -d 'trunk()'`, I don't think we want to had it rebase onto the root commit?

Reply:
We should make `rebase` a no-op when the destination is an ancestor, but other than that I think `root` is useful as a fallback. In particular so we can default the log revset to `trunk()..` (roughly) and have it show all commits in repos without a `main` or `master` branch. I think we can also change the default argument for `jj git push` be `-r 'trunk()..@'`, which I think should be safe even if we use the root commit as fallback.

Reply:
Thinking about the rebase case, we'll need to delineate between the local trunk and the remote trunk in order to ensure `trunk()..@` works. Do we need `trunk()` and `remote_trunk()` functions or should we maybe forgo `latest()` in the function so we have a full history to use?

Reply:
The trunk alias added in this PR should definitely cover points 2-4 in the short/medium term.

* I currently define `jj log` in terms of `(trunk()..@) | (trunk()..@)-` under a `jj sl` alias. There might be a better default but it will for a stack show me the entire stack plus the root. For example for this PR:
```
@  ulzolrwp ruben.slabbert 8 hours ago push-zqwpmmvnmxyx f341b78b
│  feat: add trunk alias with default to main/master/root
◉  sptkkttk martinvonz 18 hours ago HEAD@git 5f3df4aa
│  revset: resolve "@" symbol's workspace id earlier (while parsing)
~
```
* Using `remote_branches()..@` doesn't seem to work because as soon as the branch has been submit once, it can't be pushed again with that alias since it's now a remote branch as well. So it might be suitable for an initial `jj git push --change` but not for subsequent updates (which is why I have submit `git push --change @` and push `git push -r "(trunk..@):: & branches(push)"` aliases defined separately). Additionally, I've been careful about using far reaching revsets like `remote_branches()` for default pushing since I don't want to accidentally force push an unintended branch. In terms of this PR, `trunk()..@` could work well. Or potentially `trunk()..@ & mine()`. There's also a specific case here that needs to be handled where the stack is a combination of single commit and multi-commit branches that I'd need to test revsets for.
* This alias is intended to serve as the configurable rebase destination and contains the heuristic. My `jj sync` alias is `rebase -d trunk()`. I'll add some context on the commit, and I suspect adding the default changes on top of this commit would make it more clear. I've held off on making the default changes though since I suspect that would require a bit off discussion on edge cases an implications, but I'm thinking it might be worth doing to make it easier to discuss the whole change set holistically.

In terms of `trunk()` vs a set of `immutable()` commits, I suspect an `immutable()` alias would also require some plumbing throughout the rewrite commands to ensure they're not changed to match user expectation. Additionally, in the context of defaults, a user might want to treat different types of branches as immutable/protected, not just the trunk branch (e.g. release branch, or deployment branches). `trunk()` here hits the sweet spot of being immediately usable and obvious as a default in my opinion but I'm not strongly attached to that.

Reply:
@rslabbert, I sent #2135 for changing the default branches to push. Want to review that?